packages/react-tooltip-trigger/src/components/TooltipTrigger/useTooltipTrigger.ts

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps, useId } from '@fluentui/react-utilities';
+import { TooltipTriggerChildProps, TooltipTriggerProps, TooltipTriggerState } from './TooltipTrigger.types';
+import { TooltipImperativeHandle } from '../../common/TooltipProps.types';
+import { TooltipTriggerReason } from '../../common/TooltipManager.types';
+import { useTooltipContext } from '../../common/useTooltipContext';
+
+/**
+ * Names of the shorthand properties in TooltipTriggerProps
+ * {@docCategory TooltipTrigger}
+ */
+export const tooltipTriggerShorthandProps = ['tooltip'] as const;
+
+const mergeProps = makeMergeProps<TooltipTriggerState>({ deepMerge: tooltipTriggerShorthandProps });
+
+/**
+ * Create the state required to render TooltipTrigger.
+ *
+ * The returned state can be modified with hooks such as useTooltipTriggerStyles,
+ * before being passed to renderTooltipTrigger.
+ *
+ * @param props - props from this instance of TooltipTrigger
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ *
+ * {@docCategory TooltipTrigger}
+ */
+export const useTooltipTrigger = (
+  props: TooltipTriggerProps,
+  defaultProps?: TooltipTriggerProps,
+): TooltipTriggerState => {
+  const tooltipRef = React.useRef<TooltipImperativeHandle | null>(null);
+
+  const { Tooltip, manager, portalRoot } = useTooltipContext();
+
+  const state = mergeProps(
+    {
+      showDelay: 250,
+      hideDelay: 250,
+      children: props.children,
+      tooltip: {
+        as: Tooltip,
+        id: useId('tooltip-'),
+        componentRef: tooltipRef,
+        onPointerEnter: ev => manager?.onPointerEnterTooltip(ev.currentTarget),
+        onPointerLeave: ev => manager?.onPointerLeaveTooltip(ev.currentTarget),
+      },
+      manager,
+      portalRoot,
+      tooltipRef,
+    },
+    defaultProps && resolveShorthandProps(defaultProps, tooltipTriggerShorthandProps),
+    resolveShorthandProps(props, tooltipTriggerShorthandProps),
+  );
+
+  // Certain TooltipProps are on TooltipTrigger for convenience, and need to be added to state.tooltip
+  mergeProps(state, {
+    tooltip: {
+      position: state.position,
+      align: state.align,
+      subtle: state.subtle,
+      noArrow: state.noArrow,
+      offset: state.offset,
+    },
+  });
+
+  if (React.isValidElement(state.children)) {
```

Comment:
Might want to verify that there is only one child: https://reactjs.org/docs/react-api.html#reactchildrenonly

Reply:
You might also want to validate that the child is not a fragment otherwise cloning with props will break

Reply:
The type of the `children` prop helps catch some of these errors at compile time. But I see that it doesn't help with Fragments, so I'll add a check for that, and use React.Children.only to validate there's a single child.